airflow/api_fastapi/core_api/routes/ui/structure.py

```diff
@@ -63,4 +63,36 @@ def structure_data(
         "edges": edges,
     }
 
+    if external_dependencies:
+        entry_node_ref = nodes[0] if nodes else None
+        exit_node_ref = nodes[-1] if nodes else None
```

Comment:
Nodes are already topologically sorted, right?

Reply:
Exactly, there is an explicit call above when accessing the root dag group children.